lms/views/admin.py

```diff
@@ -47,20 +48,21 @@ def find_instance(self):
         if "query" not in self.request.params:
             raise HTTPBadRequest()
 
-        ai = self.application_instance_service.get(self.request.params["query"])
-        if ai:
+        try:
+            ai = self.application_instance_service.get(self.request.params["query"])
+        except ConsumerKeyError:
+            self.request.session.flash(
+                f'No application instance found for {self.request.params["query"]}',
+                "errors",
+            )
+            return HTTPFound(location=self.request.route_url("admin.instances"))
+        else:
```

Comment:
I can't make a suggestion in Github as it spans deleted lines, but this might be clearer without the look before you leap, especially as the param `query` is kind of vague.

```python
try:
    consumer_key = self.request.params["query"]
except KeyError:
    raise HTTPBadRequest()

try:
    ai = self.application_instance_service.get(consumer_key)
except ConsumerKeyError:
    self.request.session.flash(
        f'No application instance found for {consumer_key}',
        "errors",
    )
```

Reply:
Done: https://github.com/hypothesis/lms/pull/2651/commits/64d67dab9082c0d94c7aa8c9b8c958756b1ed1fc